### DISCUSSION TITLE
Issue with addChatPattern

### DISCUSSION THREAD

**Discussion by eersnington** [PrismarineJS/mineflayer#1939] [General]
```js
var regexFromMyArray = new RegExp(`(${whitelist.join("|")}): =sudo(.*)`, 'g');

bot.on("spawn", async =>{
  bot.addChatPattern('sudoPattern', regexFromMyArray, {deprecated: true})
})

bot.on('sudoPattern', (username, msg) => {
  bot.chat((String(msg).split(': =sudo'))[1])
})
```

So im trying to make a sudo command where the player can certain players in a whitelist array can make the bot say what they want by typing '=sudo {text here}' in the in game chat
but the bot sends the message twice
and when I run that command again, the bot says the first message that was sent

This is basically what happens
```
player1: =sudo hello
bot: hello
bot: hello
player1: =sudo testing
bot: hello
bot: hello
```

**Comment by Hazuki-san** [comment]
Why are you using deprecated version? You can use the new one.
or maybe... you can use bot.on('chat') with split

**Reply by amoraschi** [reply]
No, chatAddPattern is the deprecated one, he's using the right one

**Reply by Hazuki-san** [reply]
There was "{deprecated: true}" so i was thinking of something
